pdp/policy.go

```diff
@@ -243,6 +247,52 @@ func (p *Policy) delChild(ID string) (*Policy, error) {
 	return p.updatedCopy(rules, algorithm), nil
 }
 
+// MarshalWithDepth implements StorageMarshal
+func (p Policy) MarshalWithDepth(out io.Writer, depth int) error {
+	if depth < 0 {
+		return newMarshalInvalidDepthError(depth)
+	}
+	pjson, err := json.Marshal(storageNodeFmt{
+		Ord: p.ord,
+		ID:  p.id,
+	})
+	if err != nil {
+		return bindErrorf(err, "pid=\"%s\"", p.id)
+	}
+	_, err = out.Write(append(pjson[:len(pjson)-1],
```

Comment:
Append here can lead to additional allocation. It's better to rely on writer ability to buffer and just make additional call to `out.Writer([]byte(",\"rules\":["))` instead of trying to buffer here.